lib/Common/Memory/WriteBarrierMacros.h

```diff
@@ -17,6 +17,8 @@
 
 #define FieldWithBarrier(type, ...) \
     typename WriteBarrierFieldTypeTraits<type, ##__VA_ARGS__>::Type
+#define FieldWithBarrierT(type, ...) \
```

Comment:
So we will end up needing both 
`
#define FieldWithBarrier(type, ...) \
      WriteBarrierFieldTypeTraits<type, ##__VA_ARGS__>::Type
#define FieldWithBarrierT(type, ...) \
      typename WriteBarrierFieldTypeTraits<type, ##__VA_ARGS__>::Type`

for legacy builds.